lib/include/tick/solver/saga.h

```diff
@@ -43,35 +58,33 @@ class SAGA : public StoSolver {
 
   void compute_step_corrections();
 
+  void set_starting_iterate(Array<T> &new_iterate) override;
+
  public:
-  SAGA(ulong epoch_size,
-       double tol,
-       RandType rand_type,
-       double step,
-       int seed = -1,
-       VarianceReductionMethod variance_reduction = VarianceReductionMethod::Last);
+  TSAGA(ulong epoch_size, T tol, RandType rand_type, T step, int seed,
+        VarianceReductionMethod variance_reduction =
+            TSAGA<T>::VarianceReductionMethod::Last);
 
-  void solve() override;
+  virtual void solve();
```

Comment:
Why switching override to virtual?

Reply:
ah yes this is a copy paste error from another branch - it will be virtual come atomics, but not yet